pkg/registry/spec.go

```diff
@@ -0,0 +1,13 @@
+package registry
+
+import "context"
+
+type Interface interface {
+	Login(ctx context.Context) error
+	Logout(ctx context.Context) error
+	ListRepositories(ctx context.Context) ([]string, error)
+	ListTags(ctx context.Context, repository string) ([]string, error)
+	Pull(ctx context.Context, repo, tag string) error
+	Push(ctx context.Context, repo, tag string) error
+	RemoveImage(ctx context.Context, repo, tag string) error
```

Comment:
We'll need to adapt `Registry` methods to take `ctx`. But that will come in a follow up. I don't want to add extra noise in the PR.